### PR TITLE
Improve UPSERT detection

### PR DESCRIPTION
**Pull request by eolivelli**

### CODE REVIEW

**Comment by eolivelli** [comment]
@amitvc  can you please take a look ?
it is about Calcite.

We should also test HerdDB against current Calcite master.
Can you please open a PR with the update to Calcite 1.23-SNAPSHOT ?
Then we will re run it periodically

File: herddb-core/src/test/java/herddb/core/UpdateTest.java [hunk 1]
```diff
@@ -244,7 +244,8 @@ public void upsertTest() throws Exception {
             }
 
             // upsert, making n1 null now, because it has not been named in the INSERT clause
-            assertEquals(1, executeUpdate(manager, "UPSERT INTO tblspace1.tsql(k1,s1) values(?,'non-empty')", Arrays.asList("mykey")).getUpdateCount());
+            // use non uppercase casing in UPSERT keyword
+            assertEquals(1, executeUpdate(manager, "UPsert INTO tblspace1.tsql(k1,s1) values(?,'non-empty')", Arrays.asList("mykey")).getUpdateCount());
```
Comment (amitvc) [inline herddb-core/src/test/java/herddb/core/UpdateTest.java]: Is calcite able to support mix case characters now?

**Comment by amitvc** [comment]
@eolivelli Did you merge your for upgrade to 1.22 into master ?

**Comment by amitvc** [comment]
@eolivelli I will open PR from current master to sync up with calcite 1.23 snapshot. Did we release herddb to production with calcite 1.22 ?

File: herddb-core/src/test/java/herddb/core/UpdateTest.java (hunk 1 above)
Comment (eolivelli) [reply herddb-core/src/test/java/herddb/core/UpdateTest.java]: Yes it always did.
We had problems only on the jsqlparser part

**Comment by eolivelli** [comment]
Yes.
HerdDB 0.15.0 is running production.
Calcite is 1.22 in 0.15.0

**Comment by eolivelli** [comment]
@amitvc
If you are happy with this patch can you please merge it?

**Comment by eolivelli** [comment]
@amitvc ping

**Comment by eolivelli** [comment]
in order to merge you have to "approve" and then you will be able to merge the patch using the github button "squash and merge"

**Pull request merged**